Cargo.toml

```diff
@@ -129,10 +144,15 @@ sqlx = { version = "0.6", features = [
     "uuid",
 ] }
 
+typestate = "0.8.0"
+tokio-rustls = "0.26.0"
+rustls = { version = "0.23.10", features = ["aws_lc_rs"] }
+rustls-pemfile = "2.1.3"
+webpki = "0.22.4"
 tempfile = "3.3"
 portpicker = "0.1.1"
 tempdir = "0.3"
-thiserror = "1.0"
+thiserror = "1.0.63"
```

Comment:
Yeah that makes sense. I think I had added this to try and fix a cargo incompatible-version error, but it works just fine without it so I'll leave it off 👍

Reply:
@jgraettinger I'm curious what you think about [this](https://users.rust-lang.org/t/psa-please-specify-precise-dependency-versions-in-cargo-toml/71277):

> ```
> serde = "1.0.130"  # good!
> ```
> Cargo works well with versions specified this way. It is future-proof: the specified version is only the the minimum version, and still allows upgrades of minor versions. Cargo also unifies versions with the same semver-major number, so when two crates specify different versions, the higher version will be used for both.

From reading that (long...) thread, my take-away is that specifying at least minor versions (if major >= 1) is important as it prevents cargo from picking a minor version _lower_ than the one specified, while still allowing it to pick newer minor/patch (but not major) versions as needed to come up with a dependency solution. Downgrading a minor (or patch is major = 0) version is bad [because](https://semver.org/spec/v2.0.0.html#summary):

> increment the MINOR version when you add functionality in a backward compatible manner

So therefore, _decrementing_ the minor version would result in a loss of functionality that's potentially used by the code that depends on it.

Reply:
I left it at `1.0` for now